src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -271,59 +271,51 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
     source: TiDBRelation,
     dagReq: TiDAGRequest
   ): Seq[SparkPlan] = {
-    val aliasMap = mutable.HashMap[(Boolean, Expression), Alias]()
```

Comment:
Instead of building the alias map on the fly, I build it with an extra traversal over `aggregateExpressions` using an immutable map inside `toAlias`.